torch/csrc/jit/passes/guard_elimination.cpp

```diff
@@ -122,36 +129,126 @@ struct GuardElimination {
     }
   }
 
- private:
-  bool removableGuard(Node* n) {
-    if (!simple_ops_.count(n->kind())) {
-      return false;
-    }
-
+  bool checkInputs(Node *n, const std::unordered_set<size_t> &except) {
```

Comment:
What does this function do? Its name doesn't help me.

Reply:
i'll add a comment.